tests/test_predict.py

```diff
@@ -69,3 +72,48 @@ def test_predict(model_class, env_id, device):
 
         action, _ = model.predict(vec_env_obs, deterministic=False)
         assert action.shape[0] == vec_env_obs.shape[0]
+
+
+class FlattenBatchNormExtractor(BaseFeaturesExtractor):
+    """
+    Feature extract that flatten the input and uses batch normalization.
+    Used as a placeholder when feature extraction is not needed.
+
+    :param observation_space:
+    """
+
+    def __init__(self, observation_space: gym.Space, batch_norm_elements=4):
+        super(FlattenBatchNormExtractor, self).__init__(observation_space, get_flattened_obs_dim(observation_space))
+        self.flatten = nn.Flatten()
+        self.batch_norm = nn.BatchNorm1d(batch_norm_elements)
+
+    def forward(self, observations: th.Tensor) -> th.Tensor:
+        return self.batch_norm(self.flatten(observations))
+
+
+@pytest.mark.parametrize("model_class", MODEL_LIST)
+@pytest.mark.parametrize("env_id", ["Pendulum-v0", "CartPole-v1"])
+@pytest.mark.parametrize("device", ["cpu", "cuda", "auto"])
+def test_batch_norm_dqn(model_class, env_id, device):
+    if device == "cuda" and not th.cuda.is_available():
+        pytest.skip("CUDA not available")
+
+    features_extractor_kwargs = dict(batch_norm_elements=3)
+
+    if env_id == "CartPole-v1":
+        features_extractor_kwargs = dict(batch_norm_elements=4)
+
+        if model_class in [SAC, TD3]:
+            return
+    elif model_class in [DQN]:
+        return
+
+    policy_kwargs = dict(
+        features_extractor_class=FlattenBatchNormExtractor,
+        features_extractor_kwargs=features_extractor_kwargs,
+    )
+    model = model_class("MlpPolicy", env_id, policy_kwargs=policy_kwargs, verbose=1)
```

Comment:
for off-policy algorithms, you need to set `learning_starts=0` to actually test something (and you should probably use smaller network and more training steps, you can take a look at what is done with the other tests)

Reply:
Done. See d5215b69f3c6576ab54ed899a10baf85de86237a.